# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsContentUtils.h
# Commit: ca2681b0f792
# Full Hash: ca2681b0f79274712ad5f99e34075fecc6a8629c
# Author: Manuel Bucher <manuel@mozilla.com>
# Date: 2022-12-03 09:24:59
# Regressor Bug: 1771867
# File Overlap Count: 1
# Description:
#   Bug 1771867 - Early Hints Phase 2 - Part 11: Don't preload anchored links from early hints r=necko-reviewers,valentin
#   
#   They won't get passed to the PreloadService anyway. Resulting in an unnecessary
#   additional load
#   
# ==============================================================================

diff -r cbbdb4b4aaa9 -r ca2681b0f792 dom/base/nsContentUtils.h
--- a/dom/base/nsContentUtils.h	Fri Dec 02 16:15:56 2022 +0000
+++ b/dom/base/nsContentUtils.h	Fri Dec 02 16:15:57 2022 +0000
@@ -1409,6 +1409,12 @@
       nsContentPolicyType aType);
 
   /**
+   * check whether the Link header field applies to the context resource
+   * see <http://tools.ietf.org/html/rfc5988#section-5.2>
+   */
+  static bool LinkContextIsURI(const nsAString& aAnchor, nsIURI* aDocURI);
+
+  /**
    * Returns true if the content policy type is any of:
    *   * TYPE_INTERNAL_SCRIPT_PRELOAD
    *   * TYPE_INTERNAL_IMAGE_PRELOAD